### DISCUSSION TITLE
The temperature is not constant at an isothermal wall BC

### DISCUSSION THREAD

**Discussion by LDK29** [su2code/SU2#1895] [Q&A] [answered]
Hello
I am doing simulations of a hypersonic cylinder flow. I defined an isothermal BC for the cylinder. But the results show that the temperature at the cylinder is not constant. The picture below shows the temperature distribution of the cylinder.
![image](https://user-images.githubusercontent.com/113524507/213408602-caedfc75-a155-42c6-846c-b068ef8d9204.png)

Another question is the EULER BC. The Mach number near the EULER BC is not reasonable. You could find it in the picture below. I have seen some similar issues posted before, it seems still remain. So is this problem fixed in develop branch? 
![image](https://user-images.githubusercontent.com/113524507/213410537-07a590c8-d033-43fe-b539-dc23d2eedc89.png)

My configuration file is shown below. Not just for the problems I post, but any advice about my configurations is useful for me. 

`_**% ------------- DIRECT, ADJOINT, AND LINEARIZED PROBLEM DEFINITION ------------%
%
SOLVER= NEMO_NAVIER_STOKES
GAS_MODEL= AIR-5
GAS_COMPOSITION= (0.7543, 0.00713, 0.01026, 1e-5, 0.2283)
MATH_PROBLEM= DIRECT
AXISYMMETRIC= NO
RESTART_SOL= _NO

% ----------- COMPRESSIBLE AND INCOMPRESSIBLE FREE-STREAM DEFINITION ----------%
%
MACH_NUMBER= 8.98
AOA= 0.0
SIDESLIP_ANGLE= 0.0
INIT_OPTION= TD_CONDITIONS
FREESTREAM_PRESSURE= 476.0
FREESTREAM_TEMPERATURE= 901
FREESTREAM_TEMPERATURE_VE= 901
REYNOLDS_NUMBER= 10000
KIND_TURB_MODEL= NONE

% ---- NONEQUILIBRIUM GAS, IDEAL GAS, POLYTROPIC, VAN DER WAALS AND PENG ROBINSON CONSTANTS -------%
%
FLUID_MODEL= SU2_NONEQ

% -------------------- BOUNDARY CONDITION DEFINITION --------------------------%
%
MARKER_EULER= ( SYM)
MARKER_ISOTHERMAL= (CYL, 300)
MARKER_OUTLET= ( OULET, 5 )
MARKER_FAR= ( INLET )
MARKER_PLOTTING= (CYL )

%CATALYTIC_WALL = (CYL)
%SUPERCATALYTIC_WALL = YES
%SUPERCATALYTIC_WALL_COMPOSITION=(0.77,0.23,0,0,0)

MARKER_MONITORING= ( CYL )

% ------------- COMMON PARAMETERS DEFINING THE NUMERICAL METHOD ---------------i%
%
TIME_DOMAIN = NO
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES 
%(WEIGHTED_LEAST_SQUARES,GREEN_GAUSS,LEAST_SQUARES)
%
CFL_NUMBER= 0.25
CFL_ADAPT=NO
CFL_ADAPT_PARAM=(0.2,1.2,0.25,0.8,1e-6)
ITER= 100000
%LINEAR_SOLVER= BCGSTAB
%(FGMRES,BCGSTAB,...)
%LINEAR_SOLVER_ERROR= 1E-6
%LINEAR_SOLVER_ITER= 10
%LINEAR_SOLVER_PREC =LU_SGS

% -----------------------------------------------------------------------%
%
CONV_NUM_METHOD_FLOW= MSW
%(AUSM,MSW,....)
ENTROPY_FIX_COEFF= 1e-6
MUSCL_FLOW= YES
SLOPE_LIMITER_FLOW= VENKATAKRISHNAN
VENKAT_LIMITER_COEFF= 0.05
TIME_DISCRE_FLOW= EULER_EXPLICIT
%(RUNGE_KUTTA_EXPLICIT, EULER_EXPLICIT, EULER IMPLICIT)

% --------------------------- CONVERGENCE PARAMETERS --------------------------%
%
CONV_RESIDUAL_MINVAL= -25
CONV_STARTITER= 10000

% ------------------------- INPUT/OUTPUT INFORMATION --------------------------%
%
% Mesh input file
MESH_FILENAME= project3.cgns
%
% Mesh input file format (SU2, CGNS)
MESH_FORMAT= CGNS

%MESH_FILENAME= viscwedge.su2
%MESH_FORMAT= SU2
HISTORY_OUTPUT= ITER, RMS_RES
SOLUTION_FILENAME= restart_flow.dat  
TABULAR_FORMAT= TECPLOT
CONV_FILENAME= convergence
RESTART_FILENAME= restart_flow.dat
VOLUME_OUTPUT=SOLUTION,AUXILIARY, PRIMITIVE,VORTEX_IDENTIFICATION
VOLUME_FILENAME= soln_volume
SURFACE_FILENAME= soln_surface     
OUTPUT_WRT_FREQ= 2500
WRT_RESTART_OVERWRITE= YES
OUTPUT_FILES= (RESTART, PARAVIEW, SURFACE_PARAVIEW)`**_

**Comment by LDK29** [comment]
Hi
About the second question I post, I have found that is my fault for a sloping symmetry. There is a 1e-9 m difference between the two endpoints of the symmetry in y direction.
For the first question, I would be grateful if someone can help me.

**Comment by WallyMaier** [comment, marked as answer]
Hi,
Regarding the symmetry boundary, this is a known issue in the SU2 solver, and we often avoid this by simulating a full domain....I know it isn't as computationally efficient.

The Isothermal wall condition behavior seems to be OK.  The isothermal wall is a weakly enforced boundary condition, so we don't set the fluid temperature at the wall to be a certain temperature.   The saw-tooth pattern is odd, and may be due to the numerical inaccuracies caused by the symmetry plane.

**Reply by LDK29** [reply]
Hi
Many thanks for the reply. I will have a try for the full domain later.
